{
    "loaded_object_classes": "Loaded object classes.",
    "loaded_ldap_syntaxes": "Loaded LDAP syntaxes.",
    "loaded_attribute_types": "Loaded attribute types.",
    "loaded_oid_names": "Loaded object identifier names.",
    "loaded_matching_rules": "Loaded matching rules.",
    "loaded_context_types": "Loaded context types.",
    "loaded_name_forms": "Loaded name forms.",
    "transport_established": "{{source}}: {{transport}} transport established.",
    "connection_bound_auth": "{{source}}: {{protocol}} association {{aid}} bound.",
    "connection_bound_auth@with_dn": "{{source}}: {{protocol}} association {{aid}} bound. Authenticated as {{dn}}.",
    "connection_bound_anon": "{{source}}: {{protocol}} association {{aid}} bound anonymously.",
    "unsupported_protocol": "COULD_BAN={{host}}; Unsupported protocol from {{source}}: {{protocol}}.",
    "unhandled_exception": "COULD_BAN={{host}}; Unhandled exception from {{source}}: {{e}}",
    "socket_error": "COULD_BAN={{host}}; {{source}} socket error: {{e}}",
    "socket_closed": "{{source}} socked closed.",
    "server_error": "{{protocol}} server error: {{e}}",
    "listening": "{{protocol}} server listening on port {{port}}.",
    "killed_by_sentinel": "Killed by sentinel. This likely means that there was a security vulnerability discovered in Meerkat DSA. Please check the Meerkat DSA website for information. To disable this killswitch (and start up Meerkat despite there possibly being a severe security vulnerability), please restart Meerkat DSA with the MEERKAT_SENTINEL_DOMAIN environment variable set to an empty string.",
    "hibernated_by_sentinel": "Hibernation triggered by sentinel. This likely means that there was a security vulnerability discovered in Meerkat DSA. Please check the Meerkat DSA website for information. To disable this killswitch (and start up Meerkat despite there possibly being a severe security vulnerability), please restart Meerkat DSA with the MEERKAT_SENTINEL_DOMAIN environment variable set to an empty string or open the Meerkat DSA administrative console in your web browser and un-hibernate this DSA.",
    "major_version_insecure": "Your version of Meerkat DSA is insecure. Please update to the next major version.",
    "major_version_deprecated": "Your version of Meerkat DSA is no longer supported. Please consider updating to the next major version.",
    "security_update_available": "Your version of Meerkat DSA is insecure. Please update to version {{version}} or later.",
    "major_update_available": "A major update is available for Meerkat DSA.",
    "minor_update_available": "A minor update ({{version}}) is available for Meerkat DSA.",
    "patch_update_available": "A patch update ({{version}}) is available for Meerkat DSA.",
    "terminating_ob": "Terminating operational binding {{id}} at {{time}}.",
    "connection_unbound": "{{ctype}} {{cid}}: {{protocol}} association unbound.",
    "alias_but_no_aen": "Alias entry {{id}} had no aliasedEntryName attribute.",
    "subr_for_hob_not_found": "Subordinate entry for agreement {{obid}} (version {{version}}) not found.",
    "failed_to_update_hob": "Failed to update HOB for agreement {{obid}} (version {{version}}): {{e}}",
    "dsa_returned_no_opcode": "DSA '{{ae}}' returned a result with no opCode, which might have been malicious.",
    "could_not_write_operation_to_dsa": "Could not write operation to DSA '{{ae}}' because of this error: {{e}}",
    "failed_to_access_master": "Failed to access master access point {{dsa}}: {{e}}",
    "shadow_not_under_cp": "DIT invalid: shadow DSE {{id}} not under a context prefix.",
    "could_not_decode_ldap_attribute": "Could not decode LDAP attribute type {{desc}}: {{e}}",
    "failed_to_delete_old_rdn": "Failed to delete old RDN value having type {{oid}} for entry {{uuid}}.",
    "entry_deleted_while_being_modified": "Database entry {{id}} was deleted while it was being modified.",
    "loading_dit": "Loading DIT into memory. This could take a while.",
    "no_root_dse": "No root DSE found. Creating it.",
    "created_root_dse": "Created root DSE {{uuid}}.",
    "dit_loaded": "DIT loaded into memory.",
    "malformed_ldapmessage": "COULD_BAN={{host}}; Malformed LDAPMessage from client {{uuid}}.",
    "encoding_error": "COULD_BAN={{host}}; Invalid encoding from client {{uuid}}.",
    "entry_has_unrecognized_object_class": "Entry has unrecognized object class {{oid}}. It's required and optional attributes cannot be known.",
    "password_changed": "Connection {{cid}} changed the password for entry {{uuid}}.",
    "change_password_incorrect": "Connection {{cid}} supplied an incorrect password to changePassword operation on entry {{uuid}}.",
    "family_subset_had_zero_members": "Family subset had zero members. This is not supposed to ever happen.",
    "invalid_init_js": "Invalid MEERKAT_INIT_JS file. It must export a function named 'init' or 'default'.",
    "bulk_insert_mode": "You are running Meerkat DSA in bulk insert mode. This means that almost no schema validation or access controls will apply.",
    "no_dsa_keypair": "Because the MEERKAT_SIGNING_CERTS_CHAIN_FILE and MEERKAT_SIGNING_KEY environment variables were not set, Meerkat DSA will not digitally sign server responses, and its Application Entity (AE) Title will not be set.",
    "starting_meerkat_version": "Starting up Meerkat DSA, version {{version}}.",
    "starting_meerkat": "Starting up Meerkat DSA.",
    "process_info": "Process Name {{name}}; Process ID {{pid}}; Parent Process ID {{ppid}}; Current Directory: {{cwd}}.",
    "aliased_does_not_exist": "Alias {{id}} does not point to a known entry.",
    "admin_role_not_present_first_level_dse": "The administrativeRole attribute was not present in an added first-level DSE. Meerkat DSA automatically added this, making this new entry an autonomous administrative point (AAP).",
    "duplicate_name": "Duplicate name '{{name}}'. The object represented by this OID will still be usable, but not its name when using LDAP.",
    "checking_if_name_taken_in_nssr": "Checking if name is taken in NSSR via DSP operation {{opid}}.",
    "continuing_name_resolution": "Continuing name resolution via DSP operation {{opid}} for DN: '{{dn}}'",
    "resolving_name": "Resolving name '{{dn}}' for DSP operation {{opid}}.",
    "received_chained_operation": "Received chained operation with identifier {{opid}} from association {{cid}}.",
    "not_authz_to_add_top_level@hint": "To be able to add an entry to the top level in Meerkat DSA, the environment variable MEERKAT_OPEN_TOP_LEVEL must be set to '1', or there must be entries in this DSA with passwords, or you must be authenticated non-anonymously as an entry that has the boolean may_add_top_level_dse set. The first entry created in this DSA with a password will have this flag set automatically. This was the DN: {{dn}}",
    "operation_progress_not_understood": "OperationProgress value {{value}} not understood from association {{cid}}.",
    "timed_out_connecting_to_naddr": "Timed out binding to network address {{uri}}.",
    "trying_naddr": "Trying to connect to network address {{uri}}.",
    "bound_to_naddr": "Successfully bound to network address {{uri}}.",
    "attempting_starttls": "Attempting StartTLS with network address {{uri}}.",
    "starttls_established@association": "Connection {{cid}} used StartTLS to enable TLS.",
    "starttls_established@transport": "TCP connection from {{source}} used StartTLS to enable TLS.",
    "received_ldap_request": "Received LDAP request {{mid}} of type {{type}} from connection {{cid}}.",
    "received_request": "Received {{protocol}} request with invoke ID {{iid}} and operation {{op}} from connection {{cid}}.",
    "abandon_ldap_op": "Abandoning LDAP operation from message {{mid}}.",
    "abandoning_invocation_from_ldap": "Abandoning invocation {{iid}}, which corresponds to LDAP message {{mid}}.",
    "no_such_ldap_operation": "There is no LDAP operation with message ID {{mid}} for association {{assn}} in progress to abandon.",
    "tcp_max_conn": "TCP connection from {{source}} turned away because the maximum configured number of connections ({{max}}) has been reached.",
    "tcp_max_conn_per_addr": "COULD_BAN={{host}}; TCP connection from {{source}} turned away because the maximum configured number of connections ({{max}}) per address has been reached.",
    "buffer_limit": "COULD_BAN={{host}}; TCP connection from {{source}} did or would have exceeded the configured {{protocol}} buffer size of {{size}} bytes. The socket will be closed.",
    "pdu_limit": "COULD_BAN={{host}}; TCP connection from {{source}} did or would have exceeded the configured {{protocol}} PDU size of {{size}} bytes. The socket will be closed.",
    "idm_segment_limit": "COULD_BAN={{host}}; TCP connection from {{source}} did or would have exceeded the configured limit of {{limit}} IDM segments. The socket will be closed.",
    "non_ldap_data": "COULD_BAN={{host}}; TCP connection from {{source}} sent data that did not appear to be an LDAP message, judging by the first byte: 0x{{hexbyte}}.",
    "slow_loris": "COULD_BAN={{host}}; TCP connection from {{source}} sent data slower than {{bps}} bytes per second and was disconnected to prevent a Slow Loris attack.",
    "tcp_timeout": "TCP connection from {{source}} timed out after {{seconds}} seconds of inactivity and was disconnected.",
    "double_starttls": "COULD_BAN={{host}}; TCP connection from {{source}} attempted a double-StartTLS. This may have been attempted hacking.",
    "unusual_message_id": "COULD_BAN={{host}}; LDAP message from {{source}} had an unusually large or negative message ID.",
    "unusual_invoke_id": "COULD_BAN={{host}}; IDM operation from connection {{cid}} from had an unusually large or negative invoke ID.",
    "dup_invoke_id": "COULD_BAN={{host}}; Duplicated IDM invokeID {{iid}} from association {{cid}}.",
    "max_concurrent_op": "COULD_BAN={{host}}; Connection {{cid}} maxed out the number of permitted concurrent operations. The operation with invoke ID {{iid}} was rejected.",
    "address_blocked": "COULD_BAN={{host}}; Remote host {{host}} was added to the blocklist.",
    "cert_chain_no_certificates": "Certificate chain file '{{envvar}}' indicated by environment variable MEERKAT_SIGNING_CERTS_CHAIN_FILE had no PEM-encoded certificates.",
    "failed_to_decode_subordinates": "Failed to load a subordinate of entry {{uuid}} with this error. No subordinates will be cached for this entry. This was the error: {{e}}",
    "idm_bad_sequence": "COULD_BAN={{host}}; Remote host {{source}} submitted IDM packets in an invalid sequence.",
    "too_many_prebind_requests": "COULD_BAN={{host}}; Association {{cid}} submitted too many requests before receiving a bind response.",
    "bind_error": "COULD_BAN={{host}}; Error encountered during bind from host {{source}}: {{e}}",
    "admpoint_gsr_recalculated": "The governing structure rule for administrative point with dseUUID {{uuid}} was recalculated due to a change in the relevant subschema.",
    "admpoint_gsr_recalculated@hint": "Meerkat DSA does not automatically recursively recalculate the governing structure rules for subtrees of the subschema impacted by structure rule changes. Review the documentation on how to manually update the governing structure rule of the subschema.",
    "gsr_changed": "The governing structure rule for entry with dseUUID {{uuid}} changed due to a change in the relevant subschema.",
    "too_many_sort_keys": "Association {{aid}} submitted a search or list request with {{num}} sort keys, which may have been malicious. Only the first few sort keys will be honored, if any.",
    "failed_init_telemetry": "Failed to initialize telemetry. {{e}}",
    "terminating_ob@cp": "Terminating operational binding {{uuid}} because its context prefix is being deleted.",
    "terminated_ob@cp": "Terminated operational binding {{uuid}} because its context prefix is being deleted.",
    "removing_a_context_prefix": "Removing context prefix {{uuid}}.",
    "updating_superior_dsa@subentry": "Updating superior DSA because one of the context prefix's subentries were deleted.",
    "failed_to_update_superior_dsa": "Failed to update superior DSA with error {{e}}",
    "updated_superior_dsa": "Successfully updated superior DSA.",
    "establishing_hob_via_add_entry": "AddEntry.targetSystem instigated an attempt to create a HOB beneath DSE {{uuid}}.",
    "invalid_new_id": "The superior DSA indicated that the operational binding ID was already in use. This DSA will increment it and attempt it again.",
    "update_superior_dsa@oberror": "The superior DSA responded to an update with an operationalBindingError with problem {{problem}}.",
    "update_superior_dsa@errcode": "The superior DSA responded to an update with an error having error code {{code}}.",
    "update_superior_dsa@nocode": "The superior DSA responded to an update with an error without a code.",
    "establishOperationalBinding@started": "Operational binding of type {{type}} and ID {{bid}} establishment requested from association {{aid}}.",
    "modifyOperationalBinding@started": "Operational binding of type {{type}} and ID {{bid}} modification requested from association {{aid}}.",
    "terminateOperationalBinding@started": "Operational binding of type {{type}} and ID {{bid}} termination requested from association {{aid}}.",
    "establishOperationalBinding@succeeded": "Operational binding of type {{type}} and ID {{bid}} was established by association {{aid}}.",
    "modifyOperationalBinding@succeeded": "Operational binding of type {{type}} and ID {{bid}} was modified by association {{aid}}.",
    "terminateOperationalBinding@succeeded": "Operational binding of type {{type}} and ID {{bid}} was terminated by association {{aid}}.",
    "could_not_establish_connection": "Could not establish connection with DSA '{{ae}}' to chain operation {{iid}}.",
    "could_not_establish_connection@with_error": "Could not establish connection with DSA '{{ae}}' to chain operation {{iid}} because of this error: {{e}}",
    "strong_cred_error@malformed": "COULD_BAN={{host}}; Strong credentials malformed.",
    "strong_cred_error@untrusted": "COULD_BAN={{host}}; Strong credentials certification-path was not trusted.",
    "strong_cred_error@invalid_sig": "COULD_BAN={{host}}; Strong credentials had an invalid cryptographic signature.",
    "docs": "Documentation on Meerkat DSA can be found here: https://wildboar-software.github.io/directory/",
    "loaded_dsa_relationships": "Loaded DSA relationships.",
    "auth_failure_web_admin": "COULD_BAN={{host}}; Failed authentication in web admin portal.",
    "auth_success_web_admin": "Successful authentication to web admin portal from {{host}}.",
    "no_auth_web_admin": "Remote host {{host}} logged into the web admin portal without authentication, since authentication was not configured.",
    "web_admin_http": "Web admin console: {{method}} {{path}} from {{host}}",
    "anon_bind_disabled": "COULD_BAN={{host}}; Client {{host}} attempted to bind anonymously, but anonymous binds are disabled.",
    "invalid_credentials": "COULD_BAN={{host}}; Client {{host}} attempted to bind with invalid credentials.",
    "hob_invalid_entryInfo": "The potential superior DSA supplied entryInfo that describes an entry that is invalid for this reason: {{e}}.",
    "cert_path_invalid@arg": "COULD_BAN={{host}}; Invalid certification path for association {{aid}} on invocation {{iid}}.",
    "cert_path_invalid@result": "Invalid certification path in result of invocation {{iid}} from access point {{ap}}.",
    "invalid_signature@arg": "COULD_BAN={{host}}; Invalid signature for invocation {{iid}} from association {{aid}}.",
    "invalid_signature@result": "Invalid signature on result for invocation {{iid}} from access point {{ap}}.",
    "cert_path_required_signed@arg": "COULD_BAN={{host}}; Certification-path parameter is required when providing a signed argument, but was missing from invocation {{iid}} from association {{aid}}.",
    "cert_path_required_signed@result": "Certification-path parameter is required when providing a signed result, but was missing from the result of invocation {{iid}} from access point {{ap}}.",
    "cert_path_issuedToThisCA@arg": "COULD_BAN={{host}}; Certification-path CA certifications must use the issuedToThisCA field for CA certs--a constraint violated in invocation {{iid}} from association {{aid}}.",
    "cert_path_issuedToThisCA@result": "Certification-path CA certifications must use the issuedToThisCA field for CA certs--a constraint violated in invocation {{iid}} from access point {{ap}}.",
    "cert_path_issuedToThisCA@token": "Strong credentials certification-path CA certifications must use the issuedToThisCA field for CA certs.",
    "strong_cred_error@-1": "COULD_BAN={{host}}; Strong credentials invalid: Invalid signature",
    "strong_cred_error@-2": "COULD_BAN={{host}}; Strong credentials invalid: Revoked via OCSP",
    "strong_cred_error@-3": "COULD_BAN={{host}}; Strong credentials invalid: Issue reaching OCSP responder",
    "strong_cred_error@-4": "COULD_BAN={{host}}; Strong credentials invalid: Revoked via CRL",
    "strong_cred_error@-5": "COULD_BAN={{host}}; Strong credentials invalid: Issue obtaining remote CRL",
    "strong_cred_error@-6": "COULD_BAN={{host}}; Strong credentials invalid: Malformed certification path",
    "strong_cred_error@-7": "COULD_BAN={{host}}; Strong credentials invalid: Bad key usage",
    "strong_cred_error@-8": "COULD_BAN={{host}}; Strong credentials invalid: Bad extended key usage",
    "strong_cred_error@-9": "COULD_BAN={{host}}; Strong credentials invalid: Unknown critical extension",
    "strong_cred_error@-10": "COULD_BAN={{host}}; Strong credentials invalid: Duplicate extension",
    "strong_cred_error@-11": "COULD_BAN={{host}}; Strong credentials invalid: Authority Key Identifier mismatch with Subject Key Identifier",
    "strong_cred_error@-12": "COULD_BAN={{host}}; Strong credentials invalid: Signed outside of private key usage period",
    "strong_cred_error@-13": "COULD_BAN={{host}}; Strong credentials invalid: Basic constraints violated: signed by non-CA",
    "strong_cred_error@-14": "COULD_BAN={{host}}; Strong credentials invalid: Basic constraints violated: signed beyond path length constraint",
    "strong_cred_error@-15": "COULD_BAN={{host}}; Strong credentials invalid: Invalid critical extension",
    "strong_cred_error@-16": "COULD_BAN={{host}}; Strong credentials invalid: Untrusted anchor",
    "strong_cred_error@-17": "COULD_BAN={{host}}; Strong credentials invalid: Invalid time",
    "strong_cred_error@-18": "COULD_BAN={{host}}; Strong credentials invalid: Issuer-Subject mismatch",
    "strong_cred_error@-19": "COULD_BAN={{host}}; Strong credentials invalid: Name not permitted by name constraints",
    "strong_cred_error@-20": "COULD_BAN={{host}}; Strong credentials invalid: Name explicitly excluded by name constraints",
    "strong_cred_error@-21": "COULD_BAN={{host}}; Strong credentials invalid: Prohibited signature algorithm",
    "strong_cred_error@-22": "COULD_BAN={{host}}; Strong credentials invalid: Certification policy not acceptable to this DSA",
    "strong_cred_error@-23": "COULD_BAN={{host}}; Strong credentials invalid: No certification policies valid for the whole certification path",
    "idm_error": "Error with code {{code}} received from {{source}}. First few bytes of the error parameter are: {{bytes}}",
    "idm_reject": "Reject with reason {{code}} received from {{source}}.",
    "idm_abort": "Abort with reason {{code}} received from {{source}}.",
    "lcr_attempt": "Attempting to continue list operation {{iid}} for association {{aid}} in DSA {{ae}}.",
    "lcr_null_response": "No response from remote DSA {{ae}} in chained list continuation for invocation {{iid}}.",
    "lcr_error_response": "Error from from remote DSA {{ae}} in chained list continuation for invocation {{iid}}. The error code and first few bytes of the encoded error are: {{errcode}} {{errbytes}}.",
    "lcr_undefined_result": "Undefined result in response from remote DSA {{ae}} in chained list continuation for invocation {{iid}}. This is either a bug or malicious behavior.",
    "lcr_invalid_invoke_id_response": "Invalid invokeID format in response from remote DSA {{ae}} in chained list continuation for invocation {{iid}}. This is either a bug or malicious behavior.",
    "lcr_mismatch_invoke_id": "InvokeID in response from remote DSA {{ae}} in chained list continuation for invocation {{iid}} did not match the request. This is a bug. Please report it to https://github.com/Wildboar-Software/directory/issues.",
    "lcr_missing_opcode": "Missing opCode in response from remote DSA {{ae}} in chained list continuation for invocation {{iid}}. This is either a bug or malicious behavior.",
    "lcr_mismatch_opcode": "OpCode in response from remote DSA {{ae}} in chained list continuation for invocation {{iid}} was not the opCode for list. This is either a bug or malicious behavior.",
    "ldap_search": "{{cid}}#{{mid}}: LDAP search: SCOPE={{scope}}; BASE='{{base}}'; FILTER='{{filter}}'; SELECTION='{{sel}}'; TYPES_ONLY={{typesOnly}}; DEREF_ALIASES={{deref}}; SIZE_LIMIT={{size}}; TIME_LIMIT={{time}}.",
    "ldap_search_result": "{{cid}}#{{mid}}: LDAP search returned {{count}} results.",
    "acse_aare_reject": "ACSE association to {{peer}} rejected. CONTEXT={{appcontext}}; RESULT={{result}}; DIAGNOSTIC={{diag}}.",
    "acse_abrt": "ACSE association to {{peer}} aborted. SOURCE={{source}}; DIAGNOSTIC={{diag}}; ASOIID={{asoiid}}.",
    "acse_rlre_reject": "ACSE release from {{peer}} rejected. REASON={{reason}}; ASOIID={{asoiid}}.",
    "acse_unsupported_context": "ACSE application context {{appcontext}} not supported locally. Attempted ACSE association from {{peer}} will be rejected.",
    "acse_not_one_user_info": "ACSE APDU from {{peer}} did not have exactly one User-information value.",
    "acse_no_acse_user_data": "ACSE APDU from {{peer}} did not contain an extractable User-information value.",
    "presentation_abort": "Presentation-layer abort received from {{peer}}.",
    "osi_net_no_default_context_with_simply_encoded_data": "No default presentation context defined by {{peer}}, but simply-encoded-data encoding was used for a presentation data value. Association will be aborted.",
    "osi_presentation_transfer_syntax_not_supported": "Transfer syntax {{ts}} used by {{peer}} is not supported locally. Association will be aborted.",
    "osi_presentation_context_undefined": "Presentation context identifier {{pci}} was not defined in presentation association with {{peer}}. Association will be aborted.",
    "osi_pdv_multiple_transfer_syntaxes": "Presentation data value from {{peer}} sent with multiple transfer syntaxes listed. Association will be aborted.",
    "unrecognized_osi_presentation_user_data_format": "Unrecognized OSI presentation User-data format from {{peer}}. Association will be aborted.",
    "osi_net_no_pci@bind": "Bind attempted with {{peer}} using unsupported application context {{appcontext}}. No action will be taken.",
    "osi_net_no_pci@bind_result": "Bind result attempted with {{peer}} using unsupported application context {{appcontext}}. No action will be taken.",
    "osi_net_no_pci@bind_error": "Bind error attempted with {{peer}} using unsupported application context {{appcontext}}. No action will be taken.",
    "osi_net_no_pci@op": "Presentation context for operation with {{peer}} could not be determined. No action will be taken.",
    "rose_request_no_invoke_id": "ROSE request to {{peer}} enqueued with an absent Invoke-ID. An Invoke-ID of {{iid}} has been randomly chosen for this request.",
    "rose_response_no_invoke_id": "ROSE operation response (result, error, or reject) to {{peer}} enqueued with an absent Invoke-ID. This response will not be sent.",
    "osi_transport_error": "OSI transport error (ER TPDU) received from {{peer}} (DST-REF {{dst}}) with reason {{reason}}.",
    "osi_session_abort": "OSI session provider abort (AB SPDU) received from {{peer}}.",
    "osi_presentation_abort": "OSI presentation provider abort (ARP PPDU) received from {{peer}} for event ID {{eid}} and with reason code {{reason}}.",
    "will_not_bind_to_db_port": "This DSA has refused to bind to another network address to fulfill a request initiated by association {{aid}} to {{uri}} because it has the same port as the database.",
    "could_not_convert_naddr_to_url": "Could not convert N-selector (index {{i}}) to a URL for presentation address for {{aet}}.",
    "ocsp_response_invalid": "OCSP response received from {{aet}} was found to be invalid with code: {{code}}.",
    "could_not_create_rose_client": "Could not create ROSE client for connecting to {{url}}.",
    "tls_already_in_use": "TLS already in use for connecting to {{url}}.",
    "start_tls_rejected": "StartTLS request was rejected from {{url}} with reason {{reason}}.",

    "start_tls_result_received@accepted": "StartTLS acceptance received from {{url}}. Switching to TLS.",
    "start_tls_result_received@rejected": "StartTLS rejection received from {{url}} with reason {{reason}}.",
    "start_tls_abort_received": "Received abort with reason {{reason}} from {{url}} in response to StartTLS request.",
    "start_tls_timeout_received": "StartTLS request to {{url}} timed out.",
    "start_tls_other_received": "StartTLS request to {{url}} received other error: {{e}}.",
    "start_tls_not_supported_locally": "StartTLS not supported locally. No such request will be sent to {{url}}.",
    "start_tls_already_in_use": "StartTLS cannot be used, because TLS is already in use with {{url}}.",

    "bind_result_received": "Bind result received from {{url}}.",
    "bind_error_received": "Bind error received from {{url}}. First few bytes of hex encoding: {{hex}}.",
    "bind_abort_received": "Abort received from {{url}} in response to bind request.",
    "bind_timeout_received": "Bind attempt to {{url}} timed out.",
    "bind_other_received": "Bind request to {{url}} received other error: {{e}}.",

    "key_type_has_no_oid": "Could not translate key type {{kt}} to an object identifier.",
    "cannot_create_strong_creds_deps_unmet": "Cannot create strong credentials, because no signing key or cert path was defined, or the key type was not recognized.",
    "cannot_create_strong_creds_failed_pseudo_sig": "Cannot create strong credentials, because a pre-flight signature could not be produced.",
    "cannot_create_strong_creds_failed_sig": "Cannot create strong credentials, because the signature could not be produced.",
    "could_not_create_strong_creds": "Could not create strong credentials for distributed operation with {{url}}.",
    "dsa_bind_with_simple_creds": "Simple authentication attempt from {{source}}.",
    "dsa_bind_with_strong_creds": "Strong authentication attempt from {{source}}.",
    "dsa_bind_pwd_end": "Network peer {{host}} attempted to bind as DSA {{dsa}}, and their password was correct, but the remote DSA's password has expired beyond the point after which they cannot reset it themselves. Your administrative intervention may be needed to repair this broken DSA binding.",
    "dua_bind_pwd_end": "Network peer {{host}} attempted to bind, and their password was correct, but the remote user's password has expired beyond the point after which they cannot reset it themselves. Your administrative intervention may be needed to assist this user.",
    "cpw_pwd_end": "Association {{aid}} attempted to change their password, and their password was correct, but their password expired beyond the point after which they cannot reset it themselves. Your administrative intervention may be needed to assist this user.",
    "add_entry": "Association {{aid}} created new entry: DB_ID={{id}}, DSE_UUID={{uuid}}, ENTRY_UUID={{euuid}}, DN='{{dn}}'",
    "mod_entry": "Association {{aid}} modified entry: DB_ID={{id}}, DSE_UUID={{uuid}}, ENTRY_UUID={{euuid}}, DN='{{dn}}'",
    "auto_accepted_ob": "Automatically-accepted operational binding because you have MEERKAT_OB_AUTO_ACCEPT=1: OB_TYPE={{type}} OB_ID={{obid}} UUID={{uuid}}.",
    "checking_remote_pwd": "Checking remote password asserted by {{source}} via chained compare operation.",
    "error_checking_remote_pwd": "Error checking remote password asserted by {{source}} via chained compare operation: {{e}}",
    "ob_modify_attempt_by_other_ae": "There was an attempt by {{impostor}} to modify an operational binding owned by {{owner}}: OB_TYPE={{type}} OB_ID={{obid}} UUID={{uuid}}.",
    "ap_missing_ae_title": "Access point with ID {{id}} is stored in the database with a malformed AE-title. It should be an array of JSON-encoded RDNs.",
    "keylog": "SENSITIVE: TLS session with network peer {{peer}} has secret key: {{key}}",
    "scr_attempt": "Attempting to continue search operation {{iid}} for association {{aid}} in DSA {{ae}}.",
    "scr_null_response": "No response from remote DSA {{ae}} in chained search continuation for invocation {{iid}}.",
    "scr_error_response": "Error from from remote DSA {{ae}} in chained search continuation for invocation {{iid}}. The error code and first few bytes of the encoded error are: {{errcode}} {{errbytes}}.",
    "scr_undefined_result": "Undefined result in response from remote DSA {{ae}} in chained search continuation for invocation {{iid}}. This is either a bug or malicious behavior.",
    "scr_invalid_invoke_id_response": "Invalid invokeID format in response from remote DSA {{ae}} in chained search continuation for invocation {{iid}}. This is either a bug or malicious behavior.",
    "scr_mismatch_invoke_id": "InvokeID in response from remote DSA {{ae}} in chained search continuation for invocation {{iid}} did not match the request. This is a bug. Please report it to https://github.com/Wildboar-Software/directory/issues.",
    "scr_missing_opcode": "Missing opCode in response from remote DSA {{ae}} in chained search continuation for invocation {{iid}}. This is either a bug or malicious behavior.",
    "scr_mismatch_opcode": "OpCode in response from remote DSA {{ae}} in chained search continuation for invocation {{iid}} was not the opCode for search. This is either a bug or malicious behavior.",
    "scr_abstained": "Search continuation not performed. PROHIBITED_BY_USER={{prohibited}}; UNAUTHORIZED={{unauthorized}}.",
    "lcr_abstained": "List continuation not performed. PROHIBITED_BY_USER={{prohibited}}; UNAUTHORIZED={{unauthorized}}.",
    "failed_to_normalize_dn": "Failed to normalize DN: {{dn}}",
    "zero_access_points_in_cr": "Zero access points found in continuation reference for target object '{{dn}}'",
    "requester": "Requester for association {{aid}}, invocation {{iid}} is '{{r}}'.",
    "requester@search_i": "Requester for association {{aid}}, Search (I) invocation {{iid}} is '{{r}}'.",
    "requester@search_ii": "Requester for association {{aid}}, Search (II) invocation {{iid}} is '{{r}}'.",
    "search_complete": "Search complete for invocation {{iid}} by association {{aid}}.",
    "search_complete@limit": "Search limit reached for invocation {{iid}} by association {{aid}}: {{limit}}",
    "multiple_search_rules_of_same_user_class": "Detected multiple permitted and applicable search rules having user class {{user_class}}. It may be desirable to re-work the search rule definitions."
}
